src/mkdocs_autorefs/plugin.py

```diff
@@ -175,3 +239,51 @@ def on_post_page(self, output: str, page: Page, **kwargs) -> str:  # noqa: W0613
                 )
 
         return fixed_output
+
+    # Inventory fetching code
+
+    def on_pre_build(self, config: Config) -> None:  # noqa: W0613 (unused arguments)
+        """Before a build, start background tasks to download and process inventory files."""
+        if self.config["import"]:
+            inv_loader = concurrent.futures.ThreadPoolExecutor(4)
+            self._inv_futures = [
+                inv_loader.submit(self._load_inventory, handler, url)
+                for handler, urls in self.config["import"]
+                for url in urls
+            ]
+            inv_loader.shutdown(wait=False)
+
+    @classmethod
+    @functools.lru_cache(maxsize=None)
+    def _load_inventory(cls, handler, url: str) -> Mapping[str, str]:
+        """Download and process inventory files using a handler.
+
+        Arguments:
+            handler: An object responding to the method `list_object_urls`, returning a sequence of pairs.
+            url: The URL to download and process.
+
+        Returns:
+            A mapping from identifier to absolute URL.
+        """
+        log.debug(f"{__name__}: Downloading inventory from {url!r}")
+        req = urllib.request.Request(url, headers={"Accept-Encoding": "gzip"})
+        with urllib.request.urlopen(req) as resp:  # noqa: S310 (URL audit OK: comes from a checked-in config)
+            if "gzip" in resp.headers.get("content-encoding", ""):
+                resp = gzip.GzipFile(fileobj=resp)
+            result = dict(handler.list_object_urls(resp, url=url))
```

Comment:
I guess `handler.list_object_urls` is supposed to yield/return tuples like `(object_identifier, absolute_url)`?

Reply:
Yes. I also linked to a usage

> Example of how an external handler can be registered:
>
> * https://github.com/mkdocstrings/crystal/compare/inv#commit_comments_bucket

All the details are up for discussion, of course

Reply:
Yes sorry I checked that after adding this comment :smile:
Seems fine to me. Maybe pluralize "objects".